provider/datagen/src/source.rs

```diff
@@ -212,6 +227,28 @@ impl SourceData {
         &self.collations
     }
 
+    pub(crate) fn build_codepointtrie<T>(
+        &self,
+        builder: CodePointTrieBuilder<T>,
+    ) -> Result<CodePointTrie<'static, T>, DataError>
+    where
+        T: TrieValue + Into<u32>,
+    {
+        if let Some(ref path) = self.list_to_ucptrie_binary {
```

Comment:
How do you plan on testing this? Building datagen twice will have a non-trivial impact on CI times :(

Reply:
Right now? not, because I don't want to build ICU4C on CI.

Once I'm done with https://github.com/unicode-org/icu4x/issues/3121 I might add a cronjob CI for this, and also hopefully we can have something running in Google3.

Reply:
You could put the binary for the CI platforms in `datagen/tests/bin`.

Reply:
Could also stick it on GCP or something

Reply:
How big are these binaries? We already have the wasm-version in-tree, and tests should run without accessing the network.

Reply:
I'm kinda wary of checking in so many binaries into source tbh